doc/tutorials/intermediate/structure_data_store.md

```diff
@@ -0,0 +1,330 @@
+            ),
+            pn.indicators.Number(
+                value=self.data_store.avg_rotor_diameter,
+                name="Avg. Rotor Diameter (m)",
+                format="{value:,.2f}",
+                **style
+            ),
+        )
+```
+
+:::{admonition} Note
+Here we declared a `View` class that holds a reference to the `DataStore` as a parameter. Now we can have any number of concrete `View` classes that consume data from the `DataStore` and render it in any number of ways:
+:::
+
+### The App
+
+Copy the code below into a new file and save the file as `app.py`. 
+
+```python
+import param
+from panel.viewable import Viewer
+from data_store import DataStore, get_turbines
+from views import Indicators, Histogram, Table
+
+import panel as pn
+
+pn.extension("tabulator", "vega", throttled=True)
+
+class App(Viewer):
+    data_store = param.ClassSelector(class_=DataStore)
+
+    title = param.String()
+
+    views = param.List()
+
+    def __init__(self, **params):
+        super().__init__(**params)
+        updating = self.data_store.filtered.rx.updating()
+        updating.rx.watch(
+            lambda updating: pn.state.curdoc.hold()
+            if updating
+            else pn.state.curdoc.unhold()
+        )
+        self._views = pn.FlexBox(
+            *(view(data_store=self.data_store) for view in self.views), loading=updating
+        )
+        self._template = pn.template.MaterialTemplate(title=self.title)
+        self._template.sidebar.append(self.data_store)
+        self._template.main.append(self._views)
+
+    def servable(self):
+        if pn.state.served:
+            return self._template.servable()
+        return self
+
+    def __panel__(self):
+        return pn.Row(self.data_store, self._views)
+
+
+data = get_turbines()
+ds = DataStore(data=data, filters=["p_year", "p_cap", "t_manu"])
+
+App(
+    data_store=ds, views=[Indicators, Histogram, Table], title="Windturbine Explorer"
+).servable()
+```
+
+Run `panel serve app.py --autoreload`
+
+The app will look something like
+
+![Wind Turbine App with DataStore](../../_static/images/structure_data_store_app.png)
+
+## Reuse in a Notebook
+
+:::{admonition} Note
+The beauty of this compositional approach to constructing application components is that they are now usable in multiple contexts.
+:::
+
+Copy the two cells below into a notebook, remove the `#` from the imports and run the cells.
+
+```{pyodide}
+# from data_store import DataStore, get_turbines
+# from views import Indicators, Histogram, Table
+
+import panel as pn
+
+pn.extension("tabulator", "vega", throttled=True)
+```
+
+```{pyodide}
+turbines = get_turbines()
+
+ds = DataStore(data=turbines, filters=['p_year', 'p_cap', 't_manu'])
+
+pn.Row(
+    ds,
+    pn.Tabs(
+        ('Table', Table(data_store=ds)),
+        ('Histogram', Histogram(data_store=ds)),
+        ('Indicators', Indicators(data_store=ds)),
+        sizing_mode='stretch_width',
+    )
+).servable()
+```
```

Comment:
How do we get `fastparquet` installed? Right now it cannot run in Pyodide.

![image](https://github.com/holoviz/panel/assets/42288570/5ce3c649-4aa3-4071-a07f-5c1fd2209010)

Reply:
I'd say just don't depend on `fastparquet` at all, that's not the point of the tutorial.

Reply:
Yes. (I was not the one developing the example and depending on fastparquet. It was merged by phillipp :-))

Reply:
solved by using `csv.gz`